src/Node.php

```diff
@@ -388,7 +396,8 @@ public function willBeMoved()
         if ($this->isNew() || $this->willBeDeleted()) {
             return false;
         } elseif ($this->newDn !== null) {
-            return ($this->dn != $this->newDn);
+            // non-strict comparison necessary here, as we compare two objects by value
+            return $this->dn != $this->newDn; // phpcs:ignore
```

Comment:
Perhaps we can create an `equals` method on the DN-Object to make that more clear.

Reply:
I'd say that we leave it as-is to avoid magic changes/issues, for now, but we think of refactoring out `// phpcs:ignore` scenarios in future.